libs/execution/src/lib/transforms/transform-executor.ts

```diff
@@ -0,0 +1,144 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import { strict as assert } from 'assert';
+
+import {
+  InternalValueRepresentation,
+  TransformDefinition,
+  TransformOutputAssignment,
+  TransformPortDefinition,
+  Valuetype,
+  createValuetype,
+  evaluateExpression,
+} from '@jvalue/jayvee-language-server';
+
+import { ExecutionContext } from '../execution-context';
+import { TableColumn } from '../types/io-types/table';
+import { IsValidVisitor } from '../types/valuetypes/visitors/is-valid-visitor';
+
+export class TransformExecutor {
+  constructor(private readonly transform: TransformDefinition) {}
+
+  getInputDetails(): {
```

Comment:
Could be refactored to avoid code duplication, it's almost similar to `getOutputDetails`.